examples/exex/kona/src/main.rs

```diff
@@ -0,0 +1,244 @@
+            info!(target: "loop", "Validated payload attributes number {}", derived_attributes_count);
+            info!(target: "loop", "Pending l2 safe head num: {}", cursor.block_info.number);
+            match pipeline.step(cursor).await {
+                Ok(_) => info!(target: "loop", "Stepped derivation pipeline"),
+                Err(e) => warn!(target: "loop", "Error stepping derivation pipeline: {:?}", e),
+            }
+
+            if let Some(attributes) = pipeline.next_attributes() {
+                if !validator.validate(&attributes).await {
+                    error!(target: "loop", "Failed payload validation: {}", attributes.parent.block_info.hash);
+                    return Ok(());
+                }
+                derived_attributes_count += 1;
+                match l2_provider.l2_block_info_by_number(cursor.block_info.number + 1).await {
+                    Ok(bi) => {
+                        let tip = chain_provider
+                            .block_info_by_number(bi.l1_origin.number)
+                            .await
+                            .expect("Failed to fetch genesis L1 block info for pipeline tip");
+                        cursor_map.insert(tip, bi);
+                        cursor = bi;
+                    }
+                    Err(e) => {
+                        error!(target: "loop", "Failed to fetch next pending l2 safe head: {}, err: {:?}", cursor.block_info.number + 1, e);
+                    }
+                }
+                println!(
+                    "Validated Payload Attributes {derived_attributes_count} [L2 Block Num: {}] [L2 Timestamp: {}] [L1 Origin Block Num: {}]",
+                    attributes.parent.block_info.number + 1,
+                    attributes.attributes.timestamp,
+                    pipeline.origin().unwrap().number,
+                );
+                info!(target: "loop", "attributes: {:#?}", attributes);
+            } else {
+                debug!(target: "loop", "No attributes to validate");
+            }
+        } else if let Ok(s) = now.elapsed().map(|s| s.as_secs()) {
+            if s > 60 {
+                now = std::time::SystemTime::now();
+                let dist = if curr_tip <= cfg.genesis.l1.number {
+                    cfg.genesis.l1.number - curr_tip
+                } else {
+                    0
+                };
+                info!(target: "loop", "Not synced to l2 genesis. Distance to genesis: {}", dist);
+            }
+        }
+
+        if let Ok(notification) = ctx.notifications.try_recv() {
+            if let Some(reverted_chain) = notification.reverted_chain() {
+                chain_provider.commit(reverted_chain.clone());
+                let block_info = info_from_header(&reverted_chain.tip().block);
+                let blocks = reverted_chain
+                    .blocks_iter()
+                    .collect::<Vec<&reth_primitives::SealedBlockWithSenders>>();
+                for block in blocks {
```

Comment:
can you just iterate over them?

```suggestion
                for block in reverted_chain.blocks_iter() {
```